core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -278,6 +278,12 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (pyFiles != null && !isPython) {
       SparkSubmit.printErrorAndExit("--py-files given but primary resource is not a Python script")
     }
+    if (totalExecutorCores != null && executorCores != null
+        && (totalExecutorCores.toInt % executorCores.toInt) != 0) {
```

Comment:
Ok, I've modified the  mod expression repeated  to a val param and reused it,thanks.